Choropleth language independent settings

janezd opened the pull request:
##### Issue

Fixes #192. Also fixes #195.

Based on #193.

##### Description of changes

- Use aggregation name as a setting.
- FIxes for tests that fail after "tests on nasty data" were added to the widget test base.
  
##### Includes
- [X] Code changes
- [X] Tests

janezd committed:
Choropleth: Language independent settings

VesnaT reviewed (changes requested):
When I open a saved workflow, the `Show` setting does not reflect the correct state.

janezd commented:
@VesnaT, you probably mean the aggregation type, which was reset to "Count"? (It didn't work before -- see #195 --, either, but failed in a different way).

I fixed this and merged it into the previous commit. In case you already looked into the code (likely not), this is what I changed now.

```
--- a/orangecontrib/geo/widgets/owchoropleth.py
+++ b/orangecontrib/geo/widgets/owchoropleth.py
@@ -787,7 +787,7 @@ class OWChoropleth(OWWidget):
             self.clear(cache=True)
             self.input_changed.emit(data)
             self.setup_plot()
-        self.update_agg()
+        self.update_agg(from_context=True)
         self.apply_selection()
         self.unconditional_commit()

@@ -816,15 +816,22 @@ class OWChoropleth(OWWidget):
         else:
             self.agg_attr = None
         self.attr_lat, self.attr_lon = lat, lon
+        self.agg_func = DEFAULT_AGG_FUNCS.get(type(self.agg_attr),
+                                              DEFAULT_AGG_FUNC)

     def on_agg_func_changed(self, value):
         if value:
             self.agg_func = AGG_FUNCS[value].name
         self.graph.update_colors()

-    def update_agg(self):
-        # Store previous aggregation to keep it, unless it was the only choice
-        current_agg = self.agg_func_combo.count() > 1 and self.agg_func
+    def update_agg(self, *, from_context=False):
+        # This function updates the list of aggregations when the aggregation
+        # attribute is set on open context or on user interaction.
+        # If the currently selected aggregation was the only available option
+        # (thus: enforced) and we haven't just gotten this option from the context,
+        # we reset the option to the default for this attribute type.
+        current_agg = (from_context or self.agg_func_combo.count() > 1) \
+                      and self.agg_func
```

janezd committed:
Fixes for tests on nasty data, which were add to base test class
